### ISSUE TITLE
Test google sign in functionality

### ISSUE THREAD

**Issue by vidit2512** [iiitv/club_calendar#21]
### I understand and confirm the below statements

- [x] I have read the [Wiki](https://github.com/iiitv/club_calender/wiki) and I understand the concept and structure of the app.
- [x] The functionality has not already been covered in an existing issue.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within 3 days.

**Comment by vidit2512** [comment]
@iiitv/project-maintainers I want to work on this issue.

**Comment by thecodepapaya** [comment]
How do you plan to test this functionality?
What tools/frameworks are you planning to use?

**Comment by vidit2512** [comment]
I was thinking to use widget testing with mockito framework, by  which I will check that by entering valid credentials the loginpage should pump the new page by tapping login button.

**Comment by thecodepapaya** [comment]
You might want to take a look at these packages
- [firebase_auth_mocks](https://pub.dev/packages/firebase_auth_mocks)
- [google_sign_in_mocks](https://pub.dev/packages/google_sign_in_mocks)

**Comment by thecodepapaya** [comment]
Waiting for #6 to be completed first

**Comment by thecodepapaya** [comment]
@vidit2512 Would you like to take up this issue?

**Comment by vidit2512** [comment]
yes I want to work on this issue.

**Comment by thecodepapaya** [comment]
GFI @vidit2512

**Comment by thecodepapaya** [comment]
@vidit2512 do we have an update for this?

**Comment by vidit2512** [comment]
@thecodepapaya I am having a problem while accessing the login functionality as I am not able to open the login page of the app so I am not able to check my tests.

**Comment by thecodepapaya** [comment]
Can you elaborate on what kind of difficulties you are facing? Is it a problem with the app or the test which you are trying to write?
If there are any logs or unexpected behaviour associated with the app then create a separate issue for them as well.

**Comment by vidit2512** [comment]
@thecodepapaya In my test,
``` 
final AuthCredential credential = GoogleAuthProvider.credential(
        accessToken: googleAuth.accessToken,
      idToken: googleAuth.idToken,
   );
```
this snippet not working.
I have tried to google but not able to find a valid way to access google sign in mock credentials.
the same thing is discussed https://github.com/flutter/flutter/issues/42238
please let me know if you can help me,
